### ISSUE TITLE
`header-increment` and `file-name-heading` linting order

### ISSUE THREAD

**Issue by TheodorrodeohT** [platers/obsidian-linter#81]
**Problem**. When the both rules are active and the current file has no H1, then after linting the document each header increments correctly but `file-name-heading` rule does not trigger at all.

**Possible solution**. Change the execution order for the two rules so that an H1 is added first by `file-name-heading`, and then other headers are incremented by `header-increment`.

**Steps to reproduce**

1. Create a note:
    ```
    ## Header 2.1

    ## Header 2.2

    ### Header 3.1
    ```
2. Apply `header-increment` and `file-name-heading` linting rules
3. Expected result:
    ```
    # Header 1

    ## Header 2.1

    ## Header 2.2

    ### Header 3.1
    ```
4. Actual result:
    ```
    # Header 2.1

    # Header 2.2

    ## Header 3.1
    ```

**Comment by platers** [comment]
This is actually intended, I usually use header-increment when extracting sections from an old note to a new note, in which case I don't want the file name heading to be inserted. 

Perhaps there could be a forceful version of file-name-heading that inserts the heading if the current H1 heading doesnt exactly match the file name, and increments all the other headings.
